### PR TITLE
Improve code type annotations

### PR DESCRIPTION
**Pull request by Bear21**
- Remove explicit typing on "self"
- Fix load_project_epics function returning wrong epic type
- Added assert for JIRA.search_issues returning a ResultList
- add_fix_version now sends str, str as expected by add_field_value
- _load_jira_token now better handles not being able to load jiraToken

### CODE REVIEW

File: src/ittools/jira/jira_ext.py [hunk 1]
```diff
@@ -160,29 +162,29 @@ def fix_versions(self) -> List[str]:
         return list(map(lambda version: version["name"], versions))
 
     def add_fix_version(self, new_fix_version) -> None:
-        self.raw_issue.add_field_value("fixVersions", {"name": new_fix_version})
+        self.raw_issue.add_field_value("fixVersions", new_fix_version)
```
Comment (tumbarumba) [inline src/ittools/jira/jira_ext.py]: This causes a regression:

`response text = {"errorMessages":[],"errors":{"fixVersions":"Could not find valid 'id' or 'name' in version object."}}`

Comment (Bear21) [reply src/ittools/jira/jira_ext.py]: Thanks, should be resolved now.

Comment (tumbarumba) [reply src/ittools/jira/jira_ext.py]: It still shows the same error message, I'm afraid. The documentation appears to be completely wrong in this instance.

Comment (Bear21) [reply src/ittools/jira/jira_ext.py]: Would you accept bypassing this broken function?:
 ```py
      self.raw_issue.update(update={"fixVersions": [{"add": {"name": new_fix_version}}]})
```

Comment (tumbarumba) [reply src/ittools/jira/jira_ext.py]: Yep, that works

**Pull request merged**